pkg/cloud/vsphere/provisioner/common/templates.go

```diff
@@ -429,13 +435,13 @@ EOF
 echo > /etc/default/kubelet
 systemctl daemon-reload
 
-kubeadm join --token "${TOKEN}" "${MASTER}" --skip-preflight-checks --discovery-token-unsafe-skip-ca-verification
+kubeadm join --token "${TOKEN}" "${MASTER}" --discovery-token-unsafe-skip-ca-verification
```

Comment:
skip-preflight-checks was apparently removed in v1alpha3.  The error handling is not great here -- if this errors out, then the VM is left in a unprovisioned state with error messages in /var/log/cloud-init-output.log which no-one should have to read.  Still it's better to fail than continue -- my opinion.

Reply:
Theoretically, yes we should not go ahead if there is a preflight error detected by kubeadm. However, as @ntfrnzn noted, currently the VM will be left in a deserted state. 
I think an additional feature that we might want to add separately is to "build a way for the user script to be able to capture and push the errors/warnings upstream to the `Machine` object." That way if there is an error/s and/or warning that the script detects in the guest we can show that info to the user interacting with cluster-api objects. Thus the user can then go and take corrective actions.

Having said that, my opinion is until we can implement the above we should keep the `--ignore-preflight-errors=all` by default for the time being in favor of the user experience. Otherwise the users might just wait and not even know what is happening under the covers. I do support the option for the user to be able to selectively pick the error categories to ignore if they want to do that.